File: Easy/Java/1266. Minimum Time Visiting All Points.java
class Solution {
    public int minTimeToVisitAllPoints(int[][] points) {
        int xInt = points[0][0];
        int yInt = points[0][1];
        int sec = 0;

        for(int i = 0; i < points.length; i++){
            int x = points[i][0];
            int y = points[i][1];
            int dx = Math.abs(x - xInt);
            int dy = Math.abs(y - yInt);
            int temp = Math.max(dx, dy);
            dx -= temp;
            dy -= temp;
            sec += temp + Math.max(0, dx) + Math.max(0, dy);
            xInt = x;
            yInt = y;
        }

        return sec;
    }
}

File: Easy/JavaScript/1266. Minimum Time Visiting All Points.js
var minTimeToVisitAllPoints = function(points) {
    let xInt = points[0][0];
    let yInt = points[0][1];
    let sec = 0;

    for(let i = 0; i < points.length; i++){
        let x = points[i][0];
        let y = points[i][1];
        let dx = Math.abs(x - xInt);
        let dy = Math.abs(y - yInt);
        let temp = Math.max(dx, dy);
        dx -= temp;
        dy -= temp;
        sec += temp + Math.max(0, dx) + Math.max(0, dy);
        xInt = x;
        yInt = y;
    }

    return sec;
};
